OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -416,6 +416,25 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				// Move production icon sequence from Tooltip to Buildable
+				if (engineVersion < 20161018)
+				{
+					var tooltipChild = node.Value.Nodes.FirstOrDefault(n => n.Key == "Tooltip");
+					if (tooltipChild != null || (tooltipChild = node.Value.Nodes.FirstOrDefault(n => n.Key == "DisguiseToolTip")) != null)
```

Comment:
Reassigning `tooltipChild` inside the condition check is a bit dirty.  Can you change the line above to the following instead?
```csharp
var tooltipChild = node.Value.Nodes.FirstOrDefault(n => n.Key == "Tooltip")
    ?? node.Value.Nodes.FirstOrDefault(n => n.Key == "DisguiseToolTip");
```